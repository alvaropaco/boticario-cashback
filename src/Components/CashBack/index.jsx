import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles({
  root: {
    minWidth: 275,
    color: 'white',
  },
  bullet: {
    display: 'inline-block',
    margin: '0 2px',
    transform: 'scale(0.8)',
  },
  title: {
    fontSize: 14,
  },
  pos: {
    marginBottom: 12,
  },
  bgBlue: {
      background: 'linear-gradient(45deg, #2196F3 30%, #21CBF3 90%)',
      border: 0,
    borderRadius: 3,
    boxShadow: (props) =>
      props.color === 'red'
        ? '0 3px 5px 2px rgba(255, 105, 135, .3)'
        : '0 3px 5px 2px rgba(33, 203, 243, .3)',
    color: 'white',
    padding: '0 30px',
    margin: 8,
  },
});

const CashBack = () => {
  const classes = useStyles();
  const bull = <span className={classes.bullet}>â€¢</span>;

  return (
    <Card className={classes.root}>
      <CardContent className={classes.bgBlue}>
        <Typography gutterBottom variant="h5" component="h2">
            Total de Cashback acumulado
        </Typography>
        <Typography variant="h3" color="textSecondary" component="p">
            R$ 100,00
        </Typography>
      </CardContent>
    </Card>
  );
}

export default CashBack;
