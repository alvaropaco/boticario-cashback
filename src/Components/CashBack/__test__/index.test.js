import React from 'react';
import CashBack from '../';
import renderer from 'react-test-renderer';

it('renderiza component de CashBack', () => {
  const tree = renderer
    .create(<CashBack />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});