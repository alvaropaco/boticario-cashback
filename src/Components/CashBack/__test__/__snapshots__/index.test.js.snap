// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderiza component de CashBack 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root makeStyles-bgBlue-5 makeStyles-bgBlue-6"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      Total de Cashback acumulado
    </h2>
    <p
      className="MuiTypography-root MuiTypography-h3 MuiTypography-colorTextSecondary"
    >
      R$ 100,00
    </p>
  </div>
</div>
`;
